### DISCUSSION TITLE
The controlled demo is too complicated for imbeciles like me.

### DISCUSSION THREAD

**Discussion by jimmythecode** [gilbarbara/react-joyride#663] [Q&A] [answered]
I'm about to finish up on my second day of trying to figure out how to integrate React Joyride with my app. The demos are beautiful and I'm sure fantastic for all of those super talented coders out there. I'm, unfortunately, not one of them.

I've posted a question on Stack and am trying to make a demo to help anyone as hopeless as I am, but I'm finding it very difficult to isolate the useful information I need from the existing demos. I don't suppose anyone could help me by answering the queries I have made here:
https://stackoverflow.com/questions/66917872/react-joyride-steps-skipping-tooltip-not-showing-reset-not-working/66918484#66918484

**Comment by gilbarbara** [comment, marked as answer]
hey @jimmythecode 

Do you really need a controlled tour? Most cases don't need that...

> When I click 'next' the tooltip changes to a beacon, but I don't want it to and I don't understand why this happens on this step and not subsequent ones?
> It also skips Step2 (stepIndex 1). Why does it do this and how do I stop it?

Not sure why it was happening (probably because the Joyride component was rendered again between state changes).
But the callback was called twice with the `type` EVENTS.STEP_AFTER and `index` 0 so it increased the stepIndex twice. 0 -> 2

Change the conditional to:
`else if (data.index === stepIndexState && (type === EVENTS.STEP_AFTER || type === EVENTS.TARGET_NOT_FOUND))`

The beacon appeared because it skipped a step...

> When I reset my stepIndex to 0 and set run to true I would expect the tooltip to show at Step 1. Why has this not happened?

It is happening.

> When I click 'back', it doesn't go back. Joyride behaves as if I clicked 'next'.

Since you are controlling the tour you need to take the `action` in consideration.
`setStepIndexState((prev) => prev + (action === ACTIONS.PREV ? -1 : 1))`

**Reply by jimmythecode** [reply]
Ah thank you so much! I was totally lost with this.

Yes I spent yesterday seeing if I could make it work without using controlled, as I realised it had a bit more functionality than I'd anticipated. I'll post another query as to where I'm stuck on that one.
